index.html

```diff
@@ -85,88 +90,200 @@
+          the request, if it has one, and if the request is successful, SHOULD
+          NOT be visible to clients not using that transaction identifier The
+          effect of the request MUST NOT be durable unless and until the series
+          is successfully <a href="#commit">finished</a>.
+        </p>
+        <p>
+          If the request is successful, the server MUST include a
+          <code>Atomic-ID</code> response header whose value MUST be the current
+          transaction identifier.
+        </p>
+        <p>
+          If the transaction identified by the <code>Atomic-ID</code> header is
+          invalid for any reason, the server MUST NOT update any resources, and
+          it MUST respond with a <code>409 Conflict</code> HTTP status. The
+          server MAY also include more information about the reason for
+          rejecting the request in the response body.
         </p>
       </section>
 
       <section id="expiration">
         <h2>Expiration</h2>
         <p>
-          An implementation MAY enforce automatic expiration for an atomic series of requests.
-          If so, the response to any request made as part of the series MUST have a
-          <code>Atomic-Expires</code> header with the HTTP datetime at which the series will
-          expire. An implementation MAY extend the lifetime of a series upon the receipt of any
-          request specifying the unique identifier of that series in the <code>Atomic-ID</code>
-          header.
+          An implementation MAY enforce automatic expiration for an atomic
+          series of requests.  If so, the response to any request made as part
+          of the series MUST have a <code>Atomic-Expires</code> header with the
+          HTTP datetime at which the series will expire.
         </p>
+        <p>
+          An implementation SHOULD extend the lifetime of a series upon the
+          receipt of any request to resources within the scope of the
+          transaction.
+        </p>
+        <p>
+          An implementation SHOULD permit extending the lifetime of a series by
+          a <code>POST</code> request to the transaction identifier URI.
       </section>
 
       <section id="commit">
         <h2>Committing a series</h2>
         <p>
-          A client can finish an atomic series of requests by sending any
-          <code>PUT</code>/<code>POST</code>/<code>PATCH</code>/<code>DELETE</code> request
-          including both a <code>Atomic-Commit</code> header and a <code>Atomic-ID</code>
-          header with the unique identifier of an existing series. The effects of all requests
-          in the series MUST become durable and MUST become visible to all clients. The unique
-          identifier of the series MUST be expired and MUST NOT be reused.
+          A client finishes an atomic series of requests by sending a
+          <code>PUT</code> request to the transaction identifier URI.
```

Comment:
`PUT` to commit the transaction feels a bit unexpected to me. I guess the fcrepo4 approach of having a tx:commit subpath makes the intent clearer, but I understand that you were going for a single endpoint approach. So I'm not saying this has to change, but would be curious how others feel.

Reply:
I was trying to come up with the most RESTful mapping of HTTP verbs to transaction actions, and `PUT` seemed to fit the best, as long as you look at it as changing the state of the transaction from "open" to "committed". `POST` felt not like a better fit for extending the transaction, given the semantics we assign `POST` elsewhere in LDP for appending to a container. And I didn't want to overload `POST` with multiple meanings.

I wonder, would it be better if there were a minimal body specified, something like:
```
{"state": "committed"}
```
This could be mirrored by adding a `GET` request that would return:
```
{"state": "open"}
```
while the transaction was open.

Then we could specify that implementations are free to add their own JSON keys for implementation-specific purposes.

Thoughts?

Reply:
I don't mind this, although i can't think of anything else in fedora or its specification with this sort of interaction at the moment. There isn't really a way to see that a previous transaction has already been committed (other than being the one that commits it, which makes returning a state of "open" seem a little less useful

Now that I think about it, what happens if you include a tx id that has already been committed in an Atomic-Id header?

Reply:
I'm inclined to say the case of using an `Atomic-ID` of an already committed transaction would return a `409 Conflict`. That would actually also apply to the case of any transaction identifier that the server didn't recognize.

Reply:
Actually, I think that case would already be covered by this paragraph in "Adding to an atomic series":

> If the transaction identified by the <code>Atomic-ID</code> header is
> invalid for any reason, the server MUST NOT update any resources, and
> it MUST respond with a <code>409 Conflict</code> HTTP status. The
> server MAY also include more information about the reason for
> rejecting the request in the response body.

https://github.com/fcrepo/fcrepo-specification-atomic-operations/pull/8/files#diff-eacf331f0ffc35d4b482f1d15a887d3bR217-R221

Reply:
I'm not sure I like linking HTTP verbs to interactions. I feel like either adding distinct `tx:commit` & `tx:rollback` or submitting some body content. But thanks just my two cents.

Reply:
My intention was to create a RESTful interaction with a transaction resource. I agree having a body, at least for the "commit" interaction, might make that intent even clearer.